File: src/main/java/io/dustin/finstage/financial/dto/IncomeStatementResponse.java
package io.dustin.finstage.financial.dto;

import io.dustin.finstage.financial.domain.IncomeStatement;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;

@Getter
@Builder
public class IncomeStatementResponse {

    private final String symbol;
    private final LocalDate date;
    private final long revenue;
    private final long cost_of_revenue;
    private final long gross_profit;
    private final long operating_income;
    private final double eps;
    private final long net_income;

    public static IncomeStatementResponse from(IncomeStatement incomeStatement) {
        return IncomeStatementResponse.builder()
                .symbol(incomeStatement.getSymbol())
                .date(incomeStatement.getDate())
                .revenue(incomeStatement.getRevenue())
                .cost_of_revenue(incomeStatement.getCostOfRevenue())
                .gross_profit(incomeStatement.getGrossProfit())
                .operating_income(incomeStatement.getOperatingIncome())
                .eps(incomeStatement.getEps())
                .net_income(incomeStatement.getNetIncome())
                .build();
    }
}

File: src/main/kotlin/io/dustin/finstage/financial/dto/IncomeStatementResponse.kt
package io.dustin.finstage.financial.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.dustin.finstage.financial.domain.IncomeStatement
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDate

data class IncomeStatementResponse(
        @Schema(description = "기업 심볼", example = "AAPL")
        val symbol: String,

        @Schema(description = "보고일자", example = "2024-09-28")
        val date: LocalDate,

        @Schema(description = "매출", example = "391035000000")
        val revenue: Long,

        @Schema(description = "매출원가", example = "210352000000")
        val costOfRevenue: Long,

        @Schema(description = "매출총이익", example = "180683000000")
        val grossProfit: Long,

        @Schema(description = "영업이익", example = "123216000000")
        val operatingIncome: Long,

        @Schema(description = "주당순이익(EPS)", example = "6.11")
        val eps: Double,

        @Schema(description = "순이익", example = "93736000000")
        val netIncome: Long
) {
    companion object {
        fun from(incomeStatement: IncomeStatement): IncomeStatementResponse {
            return IncomeStatementResponse(
                    symbol = incomeStatement.symbol,
                    date = incomeStatement.date,
                    revenue = incomeStatement.revenue,
                    costOfRevenue = incomeStatement.costOfRevenue,
                    grossProfit = incomeStatement.grossProfit,
                    operatingIncome = incomeStatement.operatingIncome,
                    eps = incomeStatement.eps,
                    netIncome = incomeStatement.netIncome
            )
        }
    }
}
